Assets/Scripts/StatePattern/InGameDisplay.cs

```diff
@@ -124,8 +139,20 @@ void OnGUI()
                 GatherPlant();
             }
 
+            //Else the player is not at cauldron and can collect plants
```

Comment:
Wrong comment here. The player is at the cauldron and is viewing the potion canvas here.